kolena/workflow/_datatypes.py

```diff
@@ -42,6 +42,7 @@
 from kolena._utils.serde import with_serialized_columns
 from kolena._utils.validators import ValidatorConfig
 
+FIELD_ORDER_FIELD = "_field_order"
```

Comment:
It's odd to have this reserved field underscored and the `data_type` reserved field not underscored. Do you think it's worth the effort to change to `_data_type` for datasets?

Reply:
We're using the same classes as those used by current `workflow`, wouldn't that break backward compatibility?

Reply:
I think it's better to evaluate that in a separate PR.